typescript/src/timestamp.ts

```diff
@@ -25,11 +25,22 @@ export function googleProtobufTimestampToProto3JSON(
 ) {
   // seconds is an instance of Long so it won't be undefined
   const durationSeconds = obj.seconds;
-  const durationMilliseconds =
-    typeof obj.nanos === 'number' && obj.nanos > 0
-      ? Math.floor(obj.nanos / 1000000)
-      : 0;
-  return new Date(durationSeconds * 1000 + durationMilliseconds).toISOString();
+  const date = new Date(durationSeconds * 1000).toISOString();
+  let nanos = obj.nanos + '';
+  // Append leading zeros if nano string length is less than 9.
+  while (nanos.length < 9) {
+    nanos = '0' + nanos;
+  }
```

Comment:
or just `const nanos = obj.nanos.toString().padStart(9, '0')` :)

Reply:
That's super cool.